KappaAnalysis/interface/Producers/ValidMuonsProducer.h

```
@@ -238,127 +242,211 @@ class ValidMuonsProducer: public ProducerBase<TTypes>, public ValidPhysicsObject
 		}
 
 		// Apply muon isolation and MuonID
-		for (std::vector<KMuon*>::iterator muon = muons.begin(); muon != muons.end(); ++muon)
-		{
+		// DEBUG --------------------------------------
+		LOG(DEBUG) << "Apply muons iso and ID:";
+		LOG(DEBUG) << "Year: " << settings.GetYear();
+		LOG(DEBUG) << "MuonID: ";
+		if (muonID == MuonID::TIGHT) {
+			LOG(DEBUG) << "--TIGHT";
+		}
+		else if (muonID == MuonID::MEDIUM){
+			LOG(DEBUG) << "--MEDIUM";	
+		}
+		else if (muonID == MuonID::LOOSE){
+			LOG(DEBUG) << "--LOOSE";
+		}
+		else if (muonID == MuonID::VETO){
+			LOG(DEBUG) << "--VETO";
+		}
+		else {
+			LOG(DEBUG) << "--OTHER: Please verify!";
```

Shouldn't this cause a fatal?

---

This is only for information purpose.
The fatal log follows here: 
https://github.com/KIT-CMS/Artus/blob/debug_refactoring/KappaAnalysis/interface/Producers/ValidMuonsProducer.h#L333#L334

---

This is more or less "bad design", isn't it. If we know straight awy, that this is not intended, we should throw an error immediatly, imo.
But this is a code change and has nothing todo with the debug output. So maybe we can add this in another commit.

---

The error follows later on.
This output here is intended to show everything as it is configured in one place to make (config) debugging more easy.
It should not influence the logic of the actual code.
With this commit, wrong muonID, Iso or IsoType are also catched earlier!
See: https://github.com/KIT-CMS/Artus/blob/58259a41a679be0cbd875ac3b5e2e56d8a29c035/KappaAnalysis/interface/Producers/ValidMuonsProducer.h#L80 , L96, L122

---

Alright. I see. This is fine for me.

---

Also fine for me.